profiler/src/main/java/com/splunk/opentelemetry/profiler/snapshot/TraceRegistryHolder.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler.snapshot;
+
+class TraceRegistryHolder {
```

Comment:
Rest of the code uses `ConfigurableSupplier` in similar situations. I have a feeling that the existing code might be overcomplicating things we may need to revisit it at one point and see whether it could be improved.